components/server/src/workspace/env-var-service.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { ProjectDB, UserDB } from "@gitpod/gitpod-db/lib";
+import {
+    CommitContext,
+    EnvVar,
+    EnvVarWithValue,
+    ProjectEnvVar,
+    User,
+    UserEnvVar,
+    WithEnvvarsContext,
+    Workspace,
+} from "@gitpod/gitpod-protocol";
+import { inject, injectable } from "inversify";
+import { ProjectsService } from "../projects/projects-service";
+
+export interface ResolvedEnvVars {
+    user: UserEnvVar[];
+    // all project env vars, censored included always
+    project: ProjectEnvVar[];
+    context: EnvVarWithValue[];
+    // merged workspace env vars
+    workspace: EnvVar[];
```

Comment:
why do we expose the individual layers here? Isn't just returning the workspace array enough?

Reply:
Projects are used a bit differently to figure image build auth value. I added other to assert in tests. I can reduce it to workspace and projects env vars only and don't assert other layers in tests?